### PR TITLE
Antiaffinity fields for Console, KES and Prometheus

### PR DESCRIPTION
**Pull request by Alevsk**
Screen Shot 2021-08-10 at 11 03 18

Signed-off-by: Lenin Alevski <alevsk.8772@gmail.com>

### CODE REVIEW

**Review by nitisht — DISMISSED** [review]
LGTM

**Review by dvaldivia — DISMISSED** [review]

**Review by harshavardhana — APPROVED** [review]

**Review by harshavardhana — CHANGES_REQUESTED** [review]
```diff
		initContainers := []corev1.Container{
			{
				Name:  "postgres-init-chown-data",
				Image: t.Spec.Log.Db.InitImage,
				Command: []string{
					"chown",
					"-R",
					"999",
					"/var/lib/postgresql/data",
				},
				SecurityContext: &initContainerSecurityContext,
				VolumeMounts:    logVolumeMounts(t),
			},
		}
```

**Review by dvaldivia — DISMISSED** [review]

File: pkg/resources/statefulsets/log.go [hunk 1]
```diff
@@ -162,6 +164,41 @@ func NewForLogDb(t *miniov2.Tenant, serviceName string) *appsv1.StatefulSet {
...
+		// and user will have to provide a serviceAccount that allows this
+		if t.Spec.Log.Db.SecurityContext != nil {
+			var postgresRunAsUser int64
+			var postgresRunAsNonRoot bool
+			var postgresAllowPrivilegeEscalation bool
+
+			postgresRunAsUser = 0
+			postgresRunAsNonRoot = false
+			postgresAllowPrivilegeEscalation = true
+
```
Comment (harshavardhana) [inline pkg/resources/statefulsets/log.go:180]: These values are set to default `0` aren't you supposed to set this with `t.Spec.Log.Db.SecurityContext.RunAsUser` ?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: pkg/resources/statefulsets/log.go (hunk 1 above)
Comment (Alevsk) [reply pkg/resources/statefulsets/log.go:181]: Nop, this is the `initContainer` securityContext, it has to run as root to execute `chown`

**Review by dvaldivia — APPROVED** [review]

**Review by nitisht — APPROVED** [review]

**Pull request merged**